File: test/fakers/checkout/address_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/checkout/dtos/address_dto.dart';

class AddressFaker {
  static final _faker = Faker();

  static AddressDto fakeDto({
    String? id,
    String? receiver,
    String? zipcode,
    String? street,
    String? number,
    String? neighborhood,
    String? complement,
    String? city,
    String? uf,
  }) {
    return AddressDto(
      id: id ?? _faker.guid.guid(),
      receiver: receiver ?? _faker.person.name(),
      zipcode: zipcode ?? '12345-678',
      street: street ?? _faker.address.streetName(),
      number: number ?? _faker.randomGenerator.integer(1000).toString(),
      neighborhood: neighborhood ?? _faker.address.streetSuffix(),
      complement: complement ?? 'Apt ${_faker.randomGenerator.integer(100)}',
      city: city ?? _faker.address.city(),
      uf: uf ?? 'SP',
    );
  }
}

File: sertton-react-native-app/src/core/checkout/entities/fakers/address-faker.ts
import { faker } from "@faker-js/faker"

import type { Address } from "../address"

export class AddressFaker {
  static fake(address?: Partial<Address>): Address {
    return {
      city: faker.location.city(),
      complement: faker.location.secondaryAddress(),
      id: faker.string.uuid(),
      neighborhood: faker.location.county(),
      number: faker.string.numeric({ length: 3 }),
      receiver: faker.person.fullName(),
      street: faker.location.street(),
      uf: faker.location.state({ abbreviated: true }),
      zipcode: faker.location.zipCode("#####-###"),
      ...address,
    }
  }

  static fakeMany(count = 10, address?: Partial<Address>): Address[] {
    return Array.from({ length: count }, () => AddressFaker.fake(address))
  }
}
